erts/emulator/beam/erl_node_tables.c

```diff
@@ -421,8 +421,15 @@ static void schedule_delete_dist_entry(DistEntry* dep)
      *
      * Note that timeouts do not guarantee thread progress.
```

Comment:
I found this trick in erl_hl_timer.c, erts_start_timer_callback.
It is guaranteed that scheduler #1 is always online (and active), even on a system with a single core.
This is a super-rare event, because in most cases garbage collection of dist entry happens on a normal scheduler. So it does not seem necessary to take a random scheduler out of those that are online.
Making unit test for this case seems rather complicated. I'd probably suggest to have a different solution for all dirty schedulers, in erl_process.c, erts_schedule, when there is AUX WORK scheduled for dirty scheduler, either abort the emulator (because internal state is broken) or silently reschedule AUX WORK on normal scheduler.

Reply:
I think I was a bit too unclear, I want a brief comment on why this dance is done in the first place as it might not be immediately obvious. There's nothing wrong with picking scheduler 1 for this purpose.

As for the test, I think we can live without one if you haven't found a neat way to provoke this.

Reply:
After discussing it internally, we think that it's best to abort the emulator when aux work is erroneously scheduled on a dirty scheduler. Feel free to add a commit that does this.

Reply:
Ah, I just misread what you've written.
I added an explanatory comment. 
I think crashing emulator on an attempt to schedule any AUX work on a dirty scheduler should be a separate commit/PR, as it potentially touches a lot of other subsystems and may uncover even more issues.

Reply:
Great!
I'll add the assertion to `master`, we're a bit too close to the 21.2 release to include it there.